### PR TITLE
Consolidate test cases

### PR DESCRIPTION
**Pull request by cdce8p**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description
`six` is only used to test `brain_six.py`. Besides that, it isn't used anywhere. There is no reason why we can't include it with the other third party brain dependencies. Similar story for `python-dateutil`.

The tests have guards to check if `six` is installed, so there is no need for further action besides this MR.
https://github.com/PyCQA/astroid/blob/54e6efeb3989bbb1c1127a73e8e0cbc2aef78912/tests/unittest_brain.py#L63-L82

/CC: @Pierre-Sassoulas, @hippo91 

Preparation for #947

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Yeah I tried that locally too but there is a check inside the tests that six is not installed. The question then is, can we remove this check ?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
My hypothesis is that test with six and without six are required because we'd break code that do not use six. Apparently it's compatibility lib that could stay a very long time in codebase. I don't know if we can remove the tests without six. @hippo91 do you know why it was tested like this ?

**Comment by cdce8p** [comment]
> Yeah I tried that locally too but there is a check inside the tests that six is not installed. The question then is, can we remove this check ?

I would thing it should work, but I'll take a closer look.

**Comment by hippo91** [comment]
> My hypothesis is that test with six and without six are required because we'd break code that do not use six. Apparently it's compatibility lib that could stay a very long time in codebase. I don't know if we can remove the tests without six. @hippo91 do you know why it was tested like this ?

Sorry guys but have got no idea about why it has been tested like this.

**Comment by cdce8p** [comment]
I did look at this one again. At the moment, we basically have 8 tests that are run only in `six` environments. 7 of them are guarded by `HAS_SIX` and one by `HAS_DATEUTIL`. The one for `dateutil` is probably a mistake. `tox.ini` has this line here: `!py{36,37,38,39}-six: python-dateutil`. It seems as the `!` is not recognized as a logical `not` and instead just ignored.

As for the special guard in `SixBrainTest`: I believe it isn't necessary and was just a precautionary measure. There are tests in `unittest_scoped_nodes` that just check for `HAS_SIX` and they are fine.

To summarize, it should be fine to remove the `six` specific tests and just include it in `requirements_test_brain.txt`.

**Review by Pierre-Sassoulas — APPROVED** [review]
